gcc/rust/backend/rust-compile-intrinsic.cc

```diff
@@ -611,6 +614,70 @@ wrapping_op_handler_inner (Context *ctx, TyTy::FnType *fntype, tree_code op)
   return fndecl;
 }
 
+static tree
+ctpop_hander (Context *ctx, TyTy::FnType *fntype)
+{
+  rust_assert (fntype->get_params ().size () == 1);
+
+  tree lookup = NULL_TREE;
+  if (check_for_cached_intrinsic (ctx, fntype, &lookup))
+    return lookup;
+
+  auto fndecl = compile_intrinsic_function (ctx, fntype);
+
+  // setup the params
+  std::vector<Bvariable *> param_vars;
+  compile_fn_params (ctx, fntype, fndecl, &param_vars);
+
+  auto &x_param = param_vars.at (0);
+
+  if (!Backend::function_set_parameters (fndecl, param_vars))
+    return error_mark_node;
+
+  rust_assert (fntype->get_num_substitutions () == 1);
+  auto &param_mapping = fntype->get_substs ().at (0);
+  const TyTy::ParamType *param_tyty = param_mapping.get_param_ty ();
+  TyTy::BaseType *resolved_tyty = param_tyty->resolve ();
+  tree template_parameter_type
+    = TyTyResolveCompile::compile (ctx, resolved_tyty);
+
+  tree tmp_stmt = error_mark_node;
+  Bvariable *result_variable
+    = Backend::temporary_variable (fndecl, NULL_TREE, template_parameter_type,
+				   NULL_TREE, true /*address_is_taken*/,
+				   UNDEF_LOCATION, &tmp_stmt);
+
+  enter_intrinsic_block (ctx, fndecl, {result_variable});
+
+  // BUILTIN popcount FN BODY BEGIN
+  auto x = Backend::var_expression (x_param, UNDEF_LOCATION);
+
+  tree popcount_builtin = error_mark_node;
+
+  BuiltinsContext::get ().lookup_simple_builtin ("__builtin_popcountg",
```

Comment:
why `popcountg` and not `popcount` here?

Reply:
Maybe check that the builtin is correctly registered and found (rust-builtins.cc in define_builtins() and lookup_gcc_builtin.